src/screens/discover/components/NetworkSwitcher.tsx

```diff
@@ -0,0 +1,822 @@
+        x: e.x - ITEM_WIDTH * 0.5,
+        y: e.y - ITEM_HEIGHT * 0.5,
+        scale: 1.05,
+      }); // animate into the center of the pointer
+    })
+    .onChange(e => {
+      'worklet';
+      const chainId = dragging.value;
+      if (!chainId) return;
+
+      draggingTransform.modify(item => {
+        if (!item) return item;
+        item.x = e.x - ITEM_WIDTH * 0.5;
+        item.y = e.y - ITEM_HEIGHT * 0.5;
+        return item;
+      });
+
+      const isDraggingOverUnpinned = e.y > unpinnedGridY.value;
+
+      const currentIndexAtPinned = pinnedNetworks.value.indexOf(chainId);
+      const isPinned = currentIndexAtPinned !== -1;
+
+      // We don't reorder unpinned networks
+      if (isDraggingOverUnpinned && !isPinned) return;
+
+      // Unpin
+      if (isDraggingOverUnpinned && isPinned) {
+        pinnedNetworks.modify(networks => {
+          networks.splice(currentIndexAtPinned, 1);
+          return networks;
+        });
+        return;
+      }
+
+      // Pin
+      if (!isDraggingOverUnpinned && !isPinned) {
+        pinnedNetworks.modify(networks => {
+          networks.push(chainId);
+          return networks;
+        });
+        return;
+      }
+
+      // Reorder
+      const newIndex = Math.min(positionIndex(e.x, e.y), pinnedNetworks.value.length - 1);
+      if (newIndex !== currentIndexAtPinned) {
+        pinnedNetworks.modify(networks => {
+          networks.splice(currentIndexAtPinned, 1);
+          networks.splice(newIndex, 0, chainId);
+          return networks;
+        });
+      }
+    })
+    .onFinalize(e => {
+      'worklet';
+      const chainId = dragging.value;
+      if (!chainId) return;
+
+      const isDroppingInUnpinned = e.y > unpinnedGridY.value;
+
+      const index = isDroppingInUnpinned
+        ? unpinnedNetworks.value.indexOf(chainId)
+        : Math.min(positionIndex(e.x, e.y), pinnedNetworks.value.length - 1);
+
+      const { x, y } = indexPosition(index, isDroppingInUnpinned);
+
+      droppingTransform.value = draggingTransform.value;
+      droppingTransform.value = withSpring({ y, x, scale: 1 }, { mass: 0.6 }, completed => {
+        if (completed) dropping.value = null;
+        else droppingTransform.value = { y, x, scale: 1 };
+      });
+      dropping.value = dragging.value;
+      dragging.value = null;
+    })
+    .enabled(editing);
+
+  useAnimatedReaction(
+    () => unpinnedGridY.value,
+    (newY, prevY) => {
+      // the layout can recalculate after the drop started
+      if (!prevY || !droppingTransform.value) return;
+      const { x, y } = droppingTransform.value;
+      droppingTransform.value = withSpring({ x, y: y + (prevY - newY), scale: 1 }, { mass: 0.6 }, completed => {
+        if (completed) dropping.value = null;
+      });
+    }
+  );
+
+  const [isExpanded, setExpanded] = useState(false);
+
+  const toggleSelected = (chainId: ChainId) => {
+    if (selected.includes(chainId)) unselect(chainId);
+    else select(chainId);
+  };
```

Comment:
From what I understand of the spec, we only want to be able to select one network at a time (or all networks). This allows multiple to be selected.